app/src/main/addressbook.js

```diff
@@ -1,50 +1,57 @@
-const fs = require("fs-extra")
-const { join } = require("path")
-const axios = require("axios")
+let semver = require("semver")
 const url = require("url")
 
 const LOGGING = JSON.parse(process.env.LOGGING || "true") !== false
-const FIXED_NODE = process.env.COSMOS_NODE
 
 module.exports = class Addressbook {
-  constructor(
-    config,
-    configPath,
-    { persistent_peers = [], onConnectionMessage = () => {} } = {}
-  ) {
-    this.peers = []
-    this.config = config
-    this.onConnectionMessage = onConnectionMessage
-
-    // if we define a fixed node, we skip persistence
-    if (FIXED_NODE) {
-      this.addPeer(FIXED_NODE)
-      return
-    }
+  constructor(peers = [], options) {
+    Object.assign(
+      this,
+      {
+        expectedNodeVersion: undefined,
 
-    this.addressbookPath = join(configPath, "addressbook.json")
-    this.loadFromDisc()
+        // whether we should use only the single provided peer
+        fixedNode: false,
 
-    // add persistent peers to already stored peers
-    persistent_peers
-      .filter(peer => !this.peerIsKnown(peer))
-      .forEach(peer => this.addPeer(peer))
+        onConnectionMessage: () => {},
 
-    if (persistent_peers.length > 0) {
-      this.persistToDisc()
-    }
+        // used for initialization; don't pass this as an option
+        peers: [],
+
+        // a callback that persists a list of peers to disc
+        persistToDisc: () => {}
+      },
+      options
+    )
+
+    peers.forEach(peer => this.addPeer(peer))
   }
 
   async ping(peerURL) {
-    let pingURL = `http://${peerURL}:${this.config.default_tendermint_port}`
+    let pingURL = `http://${peerURL}:${
+      this.config.default_tendermint_port
+    }/status`
+
     this.onConnectionMessage(`pinging node: ${pingURL}`)
-    let nodeAlive = await axios
-      .get(pingURL, { timeout: 3000 })
-      .then(() => true, () => false)
+
+    let nodeVersion
+
+    try {
+      nodeVersion = (await this.fetch(pingURL, {
+        timeout: 3000
+      })).data.result.node_info.version
```

Comment:
Isn't that the Tendermint version?

Reply:
Ah yes, I just learned more about how Tendermint works.  Thanks, I'll fix it.

Reply:
I think your approach might not work because of this. We get the SDK version via the LCD we need to start by connecting to a node. So we can get the SDK version only after having connected to a node.

Reply:
Yes, I agree.